### PR TITLE
replacing rest-client with faraday

### PR DESCRIPTION
**Pull request by Kallin**
rest-client is no longer maintained, and contains transitive dependencies which are locked to versions behind what what's available.

use the more widely-supported faraday http-client.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -81,4 +84,4 @@ DEPENDENCIES
   rubocop-rspec
 
 BUNDLED WITH
-   1.17.3
+   2.5.17
```
Comment (Kallin) [inline Gemfile.lock:87]: this has been updated to use a modern bundler, and be compatible with ruby 3.1+. many of the transitive dependences have changed simply after running `bundle install`, but they won't affect financeit's gemfile, other than that we eventually hope to see `rest-client` disappear.

File: lib/marqeta/api_caller.rb [hunk 2]
```diff
@@ -1,61 +1,66 @@
...
 require 'marqeta/errors'
 
 module Marqeta
   class ApiCaller
     def initialize(endpoint, params = {})
       @endpoint = endpoint
-      @endpoint += "?#{URI.encode_www_form(params)}" if params.any?
+      @connection = Faraday.new(url: "http://" + Marqeta.configuration.base_url) do |conn|
+        conn.request :authorization, :basic, Marqeta.configuration.username, Marqeta.configuration.password
+        conn.response :logger, logger
```
Comment (Kallin) [inline lib/marqeta/api_caller.rb:11]: there is a [logger middleware](https://lostisland.github.io/faraday/#/middleware/included/logging) for faraday which will pass all of the request/response details directly to the logger. as a result, we don't need our manual logging calls, and don't need to test that they're happening.

**Comment by Kallin** [comment]
@dubroe , new PR for marqeta without any of the rubocop updates.

File: lib/marqeta/api_object.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 require 'marqeta/api_caller'
+require 'time'
```
Comment (dubroe) [inline lib/marqeta/api_object.rb:2]: What is this needed for?

**Review by dubroe — APPROVED** [review]
Looks good to me!

File: lib/marqeta/api_object.rb (hunk 3 above)
Comment (dubroe) [reply lib/marqeta/api_object.rb:2]: Oh interesting. I guess we don't need it for the Financeit codebase since Rails probably takes care of it.

**Pull request merged**